_shared/local_extensions.py

```diff
@@ -110,11 +112,22 @@ def include_exists(self, context, path):
     def include(self, context, path, indent=0):
         """Return the lines from the project's .cookiecutter/includes/{path} or an empty string."""
         try:
-            lines = self._readlines(context, path)
+            file_ = self._open(context, path)
```

Comment:
> `handle` or `file_handle` is a good way to avoid this.

Hmm. I know it's common term but I'm not really sure what "handle" means though. I think semantically the file object *is* the file (it's an object representing the file) rather than a handle to the file. In the same way that a date object _is_ a date not a "date handle".

These seem to be called "file objects" in the Python docs so maybe `file_object` or `file_obj`? Or just follow [the example in the docs](https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files) and use `f`?

I'll go with `file_obj`

Reply:
> Should we be doing `with self._open(context, path) as handle` here to ensure the handle is closed?

Yep, I think we should be: https://github.com/hypothesis/cookiecutters/pull/91/commits/9a54ad5c9e2e3ee537c99a551fce02735944b8a6

Reply:
Eh, whoops, that didn't work...

Reply:
Had to fix that up, I think this now works: https://github.com/hypothesis/cookiecutters/pull/91/commits/0264422ba0cc7c7c03c7240c1e51f8fa6db8e9b3